pkg/cmd/util/route.go

```diff
@@ -14,7 +14,10 @@ import (
 // UnsecuredRoute will return a route with enough info so that it can direct traffic to
 // the service provided by --service. Callers of this helper are responsible for providing
 // tls configuration, path, and the hostname of the route.
-func UnsecuredRoute(kc kclientset.Interface, namespace, routeName, serviceName, portString string) (*api.Route, error) {
+// useTargetPort choses the service TargetPort instead of Port for the route TargetPort.
+// Since the router takes over load balancing the service end points, the endpoint's port
+// is needed, not the service Port.
+func UnsecuredRoute(kc kclientset.Interface, namespace, routeName, serviceName, portString string, useTargetPort bool) (*api.Route, error) {
```

Comment:
@kargakis I have not created a test like this from scratch. Could you point to some sample code and a doc?